### PR TITLE
split the canvas into multiple files

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
The ui-jsx-canvas code is huge and a single file hosts functions that deal with very diverse concerns. This make it hard to digest the file and hard to change it without making it even longer.

**Fix:**
Move the various functions into separate files.

**Note:**
I did not change any behavior whatsoever, the code lines are also not touched. Every function is exactly the same as before. If you find any code alteration that was unintentional.

### CODE REVIEW

**Review by Rheeseyb — APPROVED** [review]
Sean is going to kill you...

File: editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+      let passthroughProps: MapLike<any> = {
+        ...assembledProps,
+        'data-uid': Utils.defaultIfNull(assembledProps['data-uid'], uid),
+        [UTOPIA_ORIGINAL_ID_KEY]: Utils.defaultIfNull(
+          assembledProps[UTOPIA_ORIGINAL_ID_KEY],
+          parentComponentInputProps[UTOPIA_ORIGINAL_ID_KEY],
+        ),
+      }
+
+      const uidForProps = Utils.defaultIfNull(assembledProps[UTOPIA_UID_KEY], uid)
```
Comment (balazsbajorics) [inline editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx:110]: @seanparsons does these look good?

File: editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+      const assembledProps = jsxAttributesToProps(
+        inScope,
+        element.props,
+        parentComponentInputProps,
+        requireResult,
+      )
+
+      let passthroughProps: MapLike<any> = {
+        ...assembledProps,
+        'data-uid': Utils.defaultIfNull(assembledProps['data-uid'], uid),
```
Comment (seanparsons) [inline editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx:103]: These shouldn't be here as they're added in the lines immediately following this.

File: editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx (hunk 1 above)
Comment (seanparsons) [reply editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx:110]: Other than the bit above, this looks great.

File: editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx (hunk 2 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx]: fixed! today I seem to have a problem seeing the difference between the red rows and the green rows :)

**Review by seanparsons — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**